tool/microkit/src/sel4.rs

```diff
@@ -410,6 +369,7 @@ impl Aarch64Regs {
         ]
     }
 
+    #[allow(clippy::len_without_is_empty)] // this will never be *empty*, bad lint
     /// Just returns the count of registers
     pub const fn len(&self) -> u64 {
```

Comment:
This should probably just be `count` instead of `len`. Do you mind changing it? Then we can get rid of this clippy ignore.

Reply:
Will do. General note for future code: It should be possible to make a `Register` trait with an associated constant `REGISTER_COUNT`, where each implementer of that trait (i. e. `Aarch64Regs`) has to provide a value for the `REGISTER_COUNT` constant.

Reply:
And I'll add a test case to verify the constant is correct.

Reply:
This is a runtime function because it is sub-optimal to return the full count of the registers. This function is used when setting up the TCB write/read register kernel invocations so you know how many registers to read/write.

I probably should have left a note/TODO, but that's why it's not a const.